src/tribol/interface/tribol.hpp

```diff
@@ -178,6 +179,62 @@ void registerMesh( integer meshId,
                    const real* y,
                    const real* z=nullptr );
 
+/*!
+ * @brief Define and register a coupling scheme over an MFEM mesh.
+ *
+ * This method is designed to enable simple registration of a contact coupling
+ * scheme over a parallel decomposed MFEM mesh.  Contact surfaces are defined
+ * via a list of boundary attributes, which is used to construct a contact
+ * surface mfem::ParSubMesh.  The parallel domains are then rebalanced using the
+ * redecomp library and the redecomp-level mesh and fields are automatically
+ * registered with Tribol.  Mesh re-decomposition is done with each call to
+ * update(), ensuring the redecomp mesh is still representative of the current
+ * state of the mesh.
+ *
+ * @param [in] cs_id Index to use for the coupling scheme.
+ * @param [in] mesh_id_1 The first ID of the contact surface mesh.
+ * @param [in] mesh_id_2 The second ID of the contact surface mesh.
+ * @param [in] mesh MFEM volume mesh.
+ * @param [in] current_coords Coordinates associated with mesh.
+ * @param [in] attributes_1 Boundary attributes defining the first mesh.
+ * @param [in] attributes_2 Boundary attributes defining the second mesh.
+ * @param [in] contact_mode 
+ * @param [in] contact_case 
+ * @param [in] contact_method 
+ * @param [in] contact_model 
+ * @param [in] enforcement_method 
+ * @param [in] binning_method 
+ */
+void registerMfemMesh( integer cs_id,
+                       integer mesh_id_1,
+                       integer mesh_id_2,
+                       mfem::ParMesh& mesh,
+                       const mfem::ParGridFunction& current_coords,
+                       const std::set<integer>& attributes_1,
+                       const std::set<integer>& attributes_2,
+                       integer contact_mode,
```

Comment:
We talked about placing all these enum options in a struct. Of course we would need a new API routine to set them all inside the struct, but then we simply pass the struct to this routine. Thoughts on that? If you are in favor, don't do for this PR, but maybe create a task/issue.

Reply:
I vote to keep it as explicit options for now.  A `struct` might help with immediate readability of the method, but it comes at the cost of obfuscation.  We can re-evaluate as we go.